lib/src/arithmetic/floating_point/floating_point_logic.dart

```diff
@@ -48,14 +48,47 @@ class FloatingPoint extends LogicStructure {
 
   /// Return a Logic true if this FloatingPoint contains a normal number,
   /// defined as having mantissa in the range [1,2)
-  Logic isNormal() => exponent.neq(LogicValue.zero.zeroExtend(exponent.width));
+  ///
+  /// IF exponent != 0 AND exponent != ALL_ONES THEN
+  Logic isNormal() =>
```

Comment:
you could make these `late final` lazy variables with names so that they only generate the signal once.  for example

```dart
late final Logic isNormal = Logic(name: 'isNormal')..gets(exponent.neq...
```

Reply:
It doesn't work. Maybe we keep it for now?

Reply:
yeah that's fine